src/python/WMCore/WMSpec/Steps/Executors/DQMUpload.py

```diff
@@ -293,3 +309,229 @@ def upload(self, url, args, filename):
+        """
+        upload  an analysis file to EOS
+        :return: boolean with upload status
+        """
+        overrides = {}
+        if hasattr(self.step, 'override'):
+            overrides = self.step.override.dictionary_()
+        eosStageOutParams = {}
+        eosStageOutParams['command'] = overrides.get('command', "xrdcp")
+        eosStageOutParams['option'] = overrides.get('option', "--wma-disablewriterecovery")
+        eosStageOutParams['phedex-node'] = overrides.get('eos-phedex-node', "T2_CH_CERN")
+        eosStageOutParams['lfn-prefix'] = overrides.get('eos-lfn-prefix',
+                                                         "root://eoscms.cern.ch/%s" % self.registerEOSPrefix)
+
+        # Switch between old and newstageOut manager.
+        # Use old by default
+        useNewStageOutCode = False
+        if 'newStageOut' in overrides and overrides.get('newStageOut'):
+            useNewStageOutCode = True
+
+        if not useNewStageOutCode:
+            # old style
+            eosmanager = StageOutMgr.StageOutMgr(**eosStageOutParams)
+            eosmanager.numberOfRetries = self.retryCount
+            eosmanager.retryPauseTime = self.retryDelay
+        else:
+            # new style
+            logging.info("DQMUpload is using new StageOut code for EOS Copy")
+            eosmanager = WMCore.Storage.FileManager.StageOutMgr(
+                    retryPauseTime=self.retryDelay,
+                    numberOfRetries=self.retryCount,
+                    **eosStageOutParams)
+
+        eosFileInfo = {'LFN': self.getAnalysisFileLFN(analysisFile),
+                       'PFN': os.path.join(stepLocation, 
+                                           os.path.basename(analysisFile.fileName)),
+                       'PNN': None,
+                       'GUID': None
+                       }
+
+        msg = "Writing DQM root files to CERN EOS with retries: %s and retry pause: %s"
+        logging.info(msg, eosmanager.numberOfRetries, eosmanager.retryPauseTime)
+        try:
+            eosmanager(eosFileInfo)
+        except Alarm:
+            msg = "Indefinite hangout while staging out to EOS"
+            logging.error(msg)
+            raise WMExecutionFailure(70317, "DQMUploadStageOutTimeout", msg)
+        except Exception as ex:
+            msg = "EOS copy failed, lfn: %s. Error: %s" % (eosFileInfo['LFN'], str(ex))
+            logging.exception(msg)
+            raise WMExecutionFailure(70318, "DQMUploadFailure", msg)
+
+        return eosFileInfo['LFN']
+
+    def getAnalysisFileLFN(self, analysisFile):
+        """
+        Construct an LFN for an analysisFile
+        """
+        # If lfn start with '/', make it relative to it
+        lfnFile = os.path.relpath(analysisFile.lfn, start='/')
+        lfn = os.path.join(self.registerLFNBase, lfnFile)
+    
+        return lfn
+
+    def _register(self, registerURL, args):
+        """
+        POST request to register URL
```

Comment:
Please update this docstring as well.

Reply:
BTW, I think we should merge this code with what we have under the `upload` method to avoid duplication. Maybe create a new method named like `_setupHttp`.

Reply:
I thought about that at first but even though both are POST requests, the `upload` method actually uploads file data content, while this doesn't, so I made them separated to distinguish the original purpose of each. What do you think?

Reply:
Maybe we could have something like `def _getHttpsOpener` with the initial common code? It it becomes too ugly, then we keep these into 2 different methods.